code/_expts/slotnick_etal_2001/de_StimCreate.m

```diff
@@ -91,21 +91,47 @@
     train.T = [];
     train.TLAB = {};
 
-    left_distances = []; % to be used when assigning labels
-    right_distances = [];
+    top_distances = []; % to be used when assigning labels
+    bottom_distances = [];
 
     % Create the 16 images
     for d1=distances % left side: distance b/n squares ranges [2, 5]
         for d2=distances % right side: distance b/n square ranges [2, 5]
-            img = paired_squares_stimuli(d1, d2, 0);
+            img = paired_squares_stimuli(d1, d2, 1);
             train.X(:, end+1) = reshape(img, prod(train.nInput), 1);
-            train.XLAB{end+1} = sprintf('(Dist) Left: %dpx ; Right: %dpx', d1, d2);
+            train.XLAB{end+1} = sprintf('(Dist) Top: %dpx ; Bottom: %dpx', d1, d2);
 
             % Coordinate task
             train.T(end+1) = (d1 == d2); % same distance or no?
             difficulty = guru_iff(d1 == d2, 'easy', 'hard');  % according to the paper...
             lbl = guru_iff(train.T(end) == 1, 'same', 'different');
             train.TLAB{end+1} = sprintf('%s-%s', lbl, difficulty);
+            
+            if d1 == d2
```

Comment:
I'm guessing this is the insertion to duplicate training examples. If so, I think the clearer way is to do:
```
train.X(:, end+1) = train.X(:, end);
train.XLAB{end+1} = train.XLAB{end}
```

Either way, please add a comment to this block about what you did and why.